flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -636,29 +636,15 @@
                         .fromCase(DOUBLE(), 0.0d, false)
                         .fromCase(DOUBLE(), -0.12345678d, true),
                 CastTestSpecBuilder.testCastTo(BINARY(2))
-                        .fromCase(CHAR(3), fromString("foo"), new byte[] {102, 111, 111})
-                        .fromCase(
-                                VARCHAR(5),
-                                fromString("Flink"),
-                                new byte[] {70, 108, 105, 110, 107})
-                        // https://issues.apache.org/jira/browse/FLINK-24419 - not trimmed to 2
-                        // bytes
-                        .fromCase(
-                                STRING(),
-                                fromString("Apache"),
-                                new byte[] {65, 112, 97, 99, 104, 101}),
+                        .fromCase(CHAR(3), StringData.fromString("foo"), new byte[] {102, 111})
```

Comment:
Please use the `fromCase` with the legacy behaviour flag and keep testing both with false and with true.